scenes/world_map/story_quest_starter.gd

```diff
@@ -31,8 +34,12 @@ func _get_configuration_warnings() -> PackedStringArray:
 ## At the end of the current interaction, enter [member quest_scene]. This is intended to be called
 ## from dialogue.
 func enter_quest() -> void:
-	interact_area.interaction_ended.connect(_do_enter_quest)
+	_enter_quest_on_dialogue_ended = true
+
 
+func _on_dialogue_ended(dialogue_resource: DialogueResource) -> void:
+	await super(dialogue_resource)
 
-func _do_enter_quest() -> void:
-	SceneSwitcher.change_to_packed(quest_scene)
+	if _enter_quest_on_dialogue_ended:
+		SceneSwitcher.change_to_packed(quest_scene)
+		_enter_quest_on_dialogue_ended = false
```

Comment:
I'm curious about why we need this boolean, is `_on_dialogue_ended` triggered more than once? Because if not, and starting a quest changes to a different scene, this boolean would be back to true when we come back to the original scene

Reply:
The Boolean is set by a particular branch of the dialogue tree calling the method above. It needs to be possible to have a branch of the dialogue that doesn't enter the quest.

Reply:
I guess the other question is, why do we need to set it back to `false` when the scene is going away anyway? Well, I suppose we don't, but IMO it captures the intent better, and if `change_to_packed` fails it might be meaningful.